### PR TITLE
Minimal Working Prototype

### PR DESCRIPTION
**Pull request by zawan-ila**
### Description:
This PR will add support for playing wordle in the terminal

### Details:
We choose a word randomly from a list of words `words.txt` and designate it as the word to be guessed. For each guess, we colour-code the output using terminal escape sequences.

### Todo:
- [ ] Add tests

### CODE REVIEW

File: wordle.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import random
+
+GREEN = '\033[32m'
+YELLOW = '\033[33m'
+DEFAULT = '\033[0m'
+
+
+def get_word(filename):
```
Comment (azanbinzahid) [inline wordle.py]: suggestion: `get_random_word or get_word(how='random')`

File: wordle.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    '''
+    Gets a random word from filename, which is a sequence of words
+    separated by newlines
+    '''
+    with open(filename, 'r') as f:
+        words = f.readlines()
+        total_words = len(words)
+        index = random.randrange(0, total_words)
```
Comment (azanbinzahid) [inline wordle.py]: also explore `random.choice()`

File: wordle.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    return words[index].strip()
+
+
+def play(word):
+    print('Guess the five letter word')
+    for attempt in range(6):
+        guess = input()
+        guess = guess[:5]
+        print('\033[A', end="")
```
Comment (azanbinzahid) [inline wordle.py]: create a variable/constant for this too.

File: wordle.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        index = random.randrange(0, total_words)
+
+    return words[index].strip()
+
+
+def play(word):
+    print('Guess the five letter word')
+    for attempt in range(6):
+        guess = input()
+        guess = guess[:5]
```
Comment (azanbinzahid) [inline wordle.py]: Allow a user to re-enter / re-use the guess if it enters a word with length not equals to 5.

File: wordle.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import random
+
+GREEN = '\033[32m'
+YELLOW = '\033[33m'
+DEFAULT = '\033[0m'
```
Comment (azanbinzahid) [inline wordle.py]: create another file to store any constants

File: wordle.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+    '''
+    with open(filename, 'r') as f:
+        words = f.readlines()
+        total_words = len(words)
+        index = random.randrange(0, total_words)
+
+    return words[index].strip()
+
+
+def play(word):
```
Comment (azanbinzahid) [inline wordle.py:58]: add docstring for method

**Review by azanbinzahid — CHANGES_REQUESTED** [review]
- Good first PR
- Address feedback changes
- Update README file with rules/instructions
- Use pep8 to auto format code locally

File: README.md [hunk 7]
```diff
@@ -1 +1,4 @@
-Wordle Game!
\ No newline at end of file
+**Wordle Game!** <br>
+Play Wordle in the terminal.
+### Instructions
+Clone the repo. Run the game using `python3 wordle.py`. Enjoy!
```
Comment (azanbinzahid) [inline README.md]: Add detailed instructions/commands (including clone command) in bullet points that a naive user can copy.

File: console_escapes.py [hunk 8]
```diff
@@ -0,0 +1,5 @@
+GREEN = '\033[32m'
```
Comment (azanbinzahid) [inline console_escapes.py]: a good naming convention would be `constants.py` for this file.

File: wordle.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+    '''
+    bad_input = False
+    guess = input()
+    err_msg = 'You should enter exactly 5 alphabetical letters'
+    while len(guess) != 5:
+        print(CURSOR_UP + LINE_CLR, end="")
+        if not bad_input:
+            print(err_msg)
+            bad_input = True
+        guess = input()
```
Comment (azanbinzahid) [inline wordle.py]: Will this accept 't0123' as correct input?

File: wordle.py [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+        for idx, letter in enumerate(guess):
+            if letter == word[idx]:
+                print(GREEN + letter + DEFAULT, end="")
+            elif letter in word:
+                print(YELLOW + letter + DEFAULT, end="")
+            else:
+                print(letter, end="")
+        print()
+
+        if guess == word:
```
Comment (azanbinzahid) [inline wordle.py]: will this handle capital/small alphabets?

File: wordle.py [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+    err_msg = 'You should enter exactly 5 alphabetical letters'
+    while len(guess) != 5:
+        print(CURSOR_UP + LINE_CLR, end="")
+        if not bad_input:
+            print(err_msg)
+            bad_input = True
+        guess = input()
+    if bad_input:
+        print(CURSOR_UP + LINE_CLR, end="")
+    print(CURSOR_UP + LINE_CLR, end="")
```
Comment (azanbinzahid) [inline wordle.py]: I see this line 3 times in this method, can we refactor?

**Review by azanbinzahid — CHANGES_REQUESTED** [review]

File: wordle.py [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+            cursor_up_line_clear(2)
+
+        print(status['err_msg'])
+        guess = input()
+
+    if bad_input:
+        cursor_up_line_clear(2)
+    else:
+        cursor_up_line_clear()
+    return guess
```
Comment (azanbinzahid) [inline wordle.py]: `cursor_up_line_clear(2 if bad_input else 1)`

File: console_escapes.py [hunk 13]
```diff
@@ -0,0 +1,5 @@
...
+YELLOW = '\033[33m'
+DEFAULT = '\033[0m'
+CURSOR_UP = '\033[A'
+LINE_CLR = '\033[2K'
```
Comment (azanbinzahid) [inline console_escapes.py]: delete this file

File: wordle.py [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+    status = {
+        'err_msg': None
+    }
+
+    while not validate(guess, status):
+        if not bad_input:
+            cursor_up_line_clear()
+            bad_input = True
+        else:
+            cursor_up_line_clear(2)
```
Comment (azanbinzahid) [inline wordle.py]:
```
cursor_up_line_clear(2 if bad_input else 1)
bad_input = True
```

File: wordle.py [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+    enter a valid guess (5 English alphabets)
+    '''
+
+    bad_input = False
+    guess = input()
+
+    # Represents the state of the guess
+    status = {
+        'err_msg': None
+    }
```
Comment (azanbinzahid) [inline wordle.py]: Why we need a dict here?

File: wordle.py (hunk 10 above)
Comment (azanbinzahid) [reply wordle.py]: allow this (and prev) to work irrespective of the string case (i.e. small, capital)

**Review by azanbinzahid — CHANGES_REQUESTED** [review]
- Please use conventional commit message 
- look how you can fixup, squash and reword commits when needed

File: wordle.py (hunk 15 above)
Comment (zawan-ila) [reply wordle.py]: Code refactored to not need a dict here.

**Review by azanbinzahid — APPROVED** [review]
Please squash commits before merging.

**Pull request merged**